src/main/kotlin/uk/gov/dluhc/printapi/config/CorrelationIdMdcConfiguration.kt

```diff
@@ -25,11 +33,14 @@ const val CORRELATION_ID_HEADER = "x-correlation-id"
  * HTTP header `x-correlation-id` if set. This allows for passing and logging a consistent correlation ID between
  * disparate systems or processes.
  */
+
 @Component
 class CorrelationIdMdcInterceptor : HandlerInterceptor {
 
     override fun preHandle(request: HttpServletRequest, response: HttpServletResponse, handler: Any): Boolean {
-        MDC.put(CORRELATION_ID, request.getHeader(CORRELATION_ID_HEADER) ?: generateCorrelationId())
+        val correlationId = request.getHeader(CORRELATION_ID_HEADER) ?: generateCorrelationId()
+        MDC.put(CORRELATION_ID, correlationId)
+        response.setHeader(CORRELATION_ID_HEADER, correlationId)
```

Comment:
It's useful to set the correlation id on the response too to aid in troubleshooting.

Reply:
Agreed. It also means the client (whatever that is - ie. our UI or some external such as EMS' in register checker) can set the same correlation ID on subsequent requests (if it makes sense to that flow/business process)